### PR TITLE
feat: add all urls to replay table

### PR DESCRIPTION
**Pull request by pauldambra**
querying replay by url is very common

we always have to go to the events table to get some event that matches the URL
which is either slow with "all events" or needs us to ask the user which event should have the filter applied

let's put the urls on the session so we can do simple querying there and prompt the user to use events when they need more

split out of https://github.com/PostHog/posthog/pull/26747 to make life easier

confirmed that locally I can ingest replay and gather all the urls in them

Screenshot 2024-12-09 at 19 21 15

### CODE REVIEW

File: plugin-server/src/main/ingestion-queues/session-recording/process-event.ts [hunk 1]
```diff
@@ -305,13 +305,10 @@ export const createSessionReplayEvent = (
                 consoleErrorCount += 1
             }
         }
-
-        if (event.type === RRWebEventType.Custom && event.data?.tag === 'Message too large') {
```
Comment (pauldambra) [inline plugin-server/src/main/ingestion-queues/session-recording/process-event.ts:309]: fly-by removal of this... rust capture doesn't support this flow anyway

File: posthog/clickhouse/test/__snapshots__/test_schema.ambr [hunk 2]
```diff
@@ -2175,7 +2177,11 @@
       distinct_id VARCHAR,
       min_first_timestamp SimpleAggregateFunction(min, DateTime64(6, 'UTC')),
       max_last_timestamp SimpleAggregateFunction(max, DateTime64(6, 'UTC')),
+      -- store the first url of the session so we can quickly show that in playlists
       first_url AggregateFunction(argMin, Nullable(VARCHAR), DateTime64(6, 'UTC')),
+      -- but also store each url so we can query by visited page without having to scan all events
+      -- despite the name we can put mobile screens in here as well to give same functionality across platforms
+      all_urls SimpleAggregateFunction(groupUniqArrayArray, Array(String)),
```
Comment (pauldambra) [inline posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: @robbie-c I realised after figuring this out that the sessions table has a URLs field too
but we've ended up with different solutions... 
is it just different inputs, am I missing a trick from something you discovered while making raw_sessions?

(i'm trying to reduce dependency on events for "basic" functionality)

Comment (robbie-c) [reply posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: Mine is also a `urls SimpleAggregateFunction(groupUniqArrayArray, Array(String)),`! Unless there's another difference you're referring to.

I don't have my own ingester though, it's coming straight from events one at a time so I just have to use what's there.

Comment (pauldambra) [reply posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: Ah, cool, i saw some array merge, array flatten that i'm not doing 

all good 👍 

(was lazy of me not to link)

Comment (pauldambra) [reply posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: https://github.com/PostHog/posthog/blob/fa716372c30050795df49cd2e6d33611c9bb5570/posthog/hogql/database/schema/sessions_v2.py#L198

Comment (pauldambra) [reply posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: hmmm great context thank you... i was hoping this was doing that for free for me... will check this before i roll out anything built on top of it!

**Review by fuziontech — APPROVED** [review]
love a good combinator

File: posthog/clickhouse/test/__snapshots__/test_schema.ambr (hunk 2 above)
Comment (robbie-c) [reply posthog/clickhouse/test/__snapshots__/test_schema.ambr:2184]: It still does a lot, most of the time! It's a big saving on bytes read, but it's just not guaranteed to happen 🙃

**Pull request merged**